# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.h
# Commit: 9896c12c4907
# Full Hash: 9896c12c490709e214030cd99f598e1ffa0076de
# Author: Peter Van der Beken <peterv@propagandism.org>
# Date: 2021-12-17 03:51:53
# Regressor Bug: 1742865
# File Overlap Count: 1
# Description:
#   Bug 1742865 - Handle meta refresh correctly with session history in the parent. r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132837
# ==============================================================================

diff -r 607f12999b71 -r 9896c12c4907 docshell/base/CanonicalBrowsingContext.h
--- a/docshell/base/CanonicalBrowsingContext.h	Thu Dec 16 15:30:29 2021 +0000
+++ b/docshell/base/CanonicalBrowsingContext.h	Thu Dec 16 15:31:56 2021 +0000
@@ -459,6 +459,12 @@
 
   void RemovePendingDiscard();
 
+  bool ShouldAddEntryForRefresh(const SessionHistoryEntry* aEntry) {
+    nsCOMPtr<nsIURI> currentURI = GetCurrentURI();
+    return BrowsingContext::ShouldAddEntryForRefresh(currentURI,
+                                                     aEntry->Info());
+  }
+
   // XXX(farre): Store a ContentParent pointer here rather than mProcessId?
   // Indicates which process owns the docshell.
   uint64_t mProcessId;